<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Version 0.3 • bcbioRNASeq</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  

<meta property="og:title" content="Version 0.3" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body>
    <div class="container template-news">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">bcbioRNASeq</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.31</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/bcbioRNASeq.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/hbc/bcbioRNASeq">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
      <h1>Changelog <small>0.3</small></h1>
      
    </div>

    <div id="bcbiornaseq-0-3-0-2018-11-06" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiornaseq-0-3-0-2018-11-06" class="anchor"></a>bcbioRNASeq 0.3.0 (2018-11-06)</h2>
<p>Development fork of <code>hbc/bcbioRNASeq</code>. Code development will proceed here on the fork until a stable release is ready to merge back at official HBC repo, to avoid any disruption for active RNA-seq experiments.</p>
<div id="new-s4-classes" class="section level3">
<h3 class="hasAnchor">
<a href="#new-s4-classes" class="anchor"></a>New S4 classes</h3>
<ul>
<li>Added new <code>DESeqAnalysis</code> S4 class, which helps containerize up a DESeq2 analysis, often requiring corresponding <code>DESeqDataSet</code>, <code>DESeqTransform</code>, and <code>DESeqResults</code> objects. Note that this class may be defined in a separate package in a future update.</li>
</ul>
</div>
<div id="code-migration-to-basejump" class="section level3">
<h3 class="hasAnchor">
<a href="#code-migration-to-basejump" class="anchor"></a>Code migration to basejump</h3>
<ul>
<li>
<code>aggregateReplicates</code> has moved to <a href="https://basejump.acidgenomics.com/">basejump</a>, and now works primarily on <code>SummarizedExperiment</code> method, instead of <code>bcbioRNASeq</code>.</li>
<li>Offloaded a number of S4 generics to basejump infrastructure, which are now defined in the <a href="https://bioverbs.acidgenomics.com/">bioverbs</a> package: <code>alphaSummary</code>, <code>contrastName</code>, <code>plot5Prime3PrimeBias</code>, <code>plotCountDensity</code>, <code>plotCountsPerGene</code>, <code>plotDEGHeatmap</code>, <code>plotDEGPCA</code>, <code>plotExonicMappingRate</code>, <code>plotGenderMarkers</code>, <code>plotGeneSaturation</code>, <code>plotGenesDetected</code>, <code>plotIntronicMappingRate</code>.</li>
<li>
<code>metrics</code> method, which returns a <code>tbl_df</code> of <code>colData</code>, has been moved to basejump.</li>
<li>
<code>plotCorrelationHeatmap</code> code is now primarily defined in basejump as a <code>SummarizedExperiment</code> method. bcbioRNASeq now calls this method internally but keeps support for <code>normalized</code> counts argument.</li>
<li>Removed <code>tpm</code> method support, which is now handled by <code>SummarizedExperiment</code> method support in basejump.</li>
</ul>
</div>
<div id="s4-method-reworks" class="section level3">
<h3 class="hasAnchor">
<a href="#s4-method-reworks" class="anchor"></a>S4 method reworks</h3>
<ul>
<li>Reworked internal code, splitting out S4 methods into internal functions (e.g. <code>counts.bcbioRNASeq</code>). This step is recommended by <a href="https://bioconductor.org/">Bioconductor</a>, and can make troubleshooting easier.</li>
<li>
<code>export</code>: Reworked S4 methods for <code>DESeqResults</code> and <code>DESeqResultsTables</code> S4 classes.</li>
<li>
<code>extract</code> (i.e. <code><a href="https://rdrr.io/r/base/Extract.html">[</a></code>) method: Switched <code>transform</code> formal to <code>recalculate</code>. Added additional assert checks. Operation no longer changes the saved <code>version</code> of bcbioRNASeq, slotted into <code>metadata</code>.</li>
<li>
<code>interestingGroups</code> formal is now set <code>NULL</code> instead of missing for plotting functions.</li>
<li>
<code>markdown</code>: Initial support of new S4 generic, which returns a markdown table of sample metadata.</li>
<li>
<code>plot5Prime3PrimeBias</code>: Added an addition grep matching step to handle differential variations on the <code>x5x3Bias</code> column in <code>colData</code>.</li>
<li>
<code>plotCountDensity</code> S4 method has been removed. Use <code>plotGeneSaturation</code> generic with <code>geom = "density"</code> argument instead.</li>
<li>
<code>plotCountsPerBiotype</code>: New S4 method support, which calls basejump <code>SummarizedExperiment</code> method.</li>
<li>
<code>plotDEGHeatmap</code>: Draft update of <code>DESeqResults</code> method.</li>
<li>
<code>plotDEGPCA</code>: Draft update of <code>DESeqResults</code> method.</li>
<li>
<code>plotDispEsts</code>: Split out and reworked internal method definition.</li>
<li>
<code>plotGenesDetected</code>: This code has been moved to basejump and is now defined against <code>SummarizedExperiment</code>.</li>
<li>
<code>plotMA</code>: Reworked internal <code>DESeqResults</code> method code. Added draft support for new <code>DESeqAnalysis</code> S4 class. Keeping <code>plotMeanAverage</code> reexported, since this function is used in the F1000 workflow paper.</li>
<li>
<code>plotMappingRate</code>: Set the default limit back down to 0.7.</li>
<li>
<code>plotMeanSD</code>: Reworked internal <code>DESeqDataSet</code> code. Added additional assert checks and changed the default handling of <code>vst</code> and <code>rlog</code> internal formals.</li>
<li>
<code>plotPCA</code>: Draft update of <code>SummarizedExperiment</code>. Will offload this code to basejump in a future update. Switched <code>return</code> argument to support <code>tibble</code> and S4 <code>DataFrame</code> – note change from previous support for <code>data.frame</code>.</li>
<li>Deleted <code>plotPCACovariates</code> method, since DEGreport is failing. Will add back support for this generic in a future update.</li>
<li>
<code>plotQC</code>: Initial method support for <code>bcbioRNASeq</code>.</li>
<li>
<code>plotTotalReads</code>: Reworked and tightened up internal ggplot2 code.</li>
<li>
<code>plotVolcano</code>: Draft update to <code>DESeqResults</code> method. Will rework this code with a <code>DESeqAnalysis</code> approach in a future update.</li>
<li>
<code>relativeLogExpression</code>: Added S4 method support.</li>
<li>
<code>resultsTables</code>: Removed this approach to subsetting <code>DESeqResults</code>. This will be a rework in the upcoming <a href="https://deseqanalysis.acidgenomcis.com/">DESeqAnalysis</a> package.</li>
<li>
<code>sampleData</code>: Simplified internal code, which takes advantage of updates to <code>SummarizedExperiment</code> defined in basejump.</li>
<li>
<code>updateObject</code>: Reworked internal metadata handling.</li>
</ul>
</div>
<div id="example-data-reorganization" class="section level3">
<h3 class="hasAnchor">
<a href="#example-data-reorganization" class="anchor"></a>Example data reorganization</h3>
<ul>
<li>Renamed and simplified example datasets. Now only exporting <code>bcb</code> (<code>bcbioRNASeq</code>) and <code>deseq</code> (<code>DESeqDataSet</code>) examples.</li>
<li>
<code>bcb_small</code>, <code>dds_small</code>, <code>gender_markers</code>, and <code>res_small</code> have been removed.</li>
<li>Updated and consolidated corresponding <code>data-raw/</code> scripts.</li>
<li>Reworked internal <code>extdata/bcbio/</code> example datasets. Testing out both <code>2017-05-23_rnaseq</code> and <code>2018-03-18_GSE67267-merged</code> data.</li>
</ul>
</div>
<div id="r-markdown-templates" class="section level3">
<h3 class="hasAnchor">
<a href="#r-markdown-templates" class="anchor"></a>R Markdown templates</h3>
<ul>
<li>Stripped down supported R Markdown templates. Currently simplified code base to support Quality Control markdown (<code>quality_control</code>). Differential expression and functional analysis templates have been temporarily removed but will be re-added in the future.</li>
</ul>
</div>
<div id="minor-changes-31" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-31" class="anchor"></a>Minor changes</h3>
<ul>
<li>Reorganized S4 class definitions and validity checks in <code>AllClasses.R</code>. Previously, this code was split out per S4 (e.g. <code>bcbioRNASeq-validity.R</code>).</li>
<li>Moved S4 coercion methods from <code>coerce-methods.R</code> to <code>as-methods.R</code>.</li>
<li>Added internal assert checks in <code>assert-internal.R</code>.</li>
<li>Consolidated internal DESeq2 code into <code>DESeq2-internal.R</code>.</li>
<li>Consolidated and defined additional global variables into <code>globals.R</code>.</li>
<li>Reorganized internal <code>.meltCounts</code> code.</li>
<li>Consolidated global package parameter arguments into <code>params.R</code>.</li>
<li>
<code>bcbio</code> prefix has been renamed to <code>basejump</code>, where applicable. Refer to global function params defined with <code>getOption</code>, and internal ggplot2 utility functions, such as <code>basejump_geom_abline</code>.</li>
<li>Draft support for <code>countsFromAbundance</code> selection in internal <code>tximport</code> call. This is useful for some transcript-level analyses.</li>
</ul>
</div>
<div id="deprecations" class="section level3">
<h3 class="hasAnchor">
<a href="#deprecations" class="anchor"></a>Deprecations</h3>
<ul>
<li>Removed defunct <code>download</code> function.</li>
<li>New deprecations: <code>plotCountDensity</code>, <code>plotPCACovariates</code>, <code>resultsTables</code>.</li>
</ul>
</div>
</div>
    <div id="bcbiornaseq-0-3-1-2018-11-14" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiornaseq-0-3-1-2018-11-14" class="anchor"></a>bcbioRNASeq 0.3.1 (2018-11-14)</h2>
<div id="major-changes-13" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes-13" class="anchor"></a>Major changes</h3>
<ul>
<li>Improved validity method for <code>bcbioRNASeq</code> S4 class.</li>
<li>
<code>extract</code> (<code><a href="https://rdrr.io/r/base/Extract.html">[</a></code>) method: Improved internal handling of <code>rowRanges</code> and <code>colData</code>. Now using <code>I</code> internally for complex S4 columns in <code>mcols</code> for <code>rowRanges</code>.</li>
<li>Reworked imports back to basejump instead of attempting to use subpackages: basejump.annotations, basejump.assertions, basejump.classes, basejump.coerion, basejump.developer, basejump.experiment, basejump.generics, basejump.globals, basejump.io, basejump.markdown, basejump.plots, basejump.sanitization. Rethink this approach in a future update.</li>
</ul>
</div>
<div id="minor-changes-30" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-30" class="anchor"></a>Minor changes</h3>
<ul>
<li>Resaved example <code>bcb</code> dataset.</li>
<li>Miscellaneous internal assert check fixes and updates.</li>
<li>Miscellaneous documentation improvements.</li>
<li>Added internal <code>Rle</code> global variable. May want to avoid this approach because <code>Rle</code> is a defined function in S4Vectors, however.</li>
<li>Offloaded some global documentation params to basejump.</li>
<li>Reworked some global params set by <code>getOption</code> in formals.</li>
</ul>
</div>
</div>
    <div id="bcbiornaseq-0-3-2-2018-11-19" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiornaseq-0-3-2-2018-11-19" class="anchor"></a>bcbioRNASeq 0.3.2 (2018-11-19)</h2>
<div id="major-changes-12" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes-12" class="anchor"></a>Major changes</h3>
<ul>
<li>Splitting out new <code>DESeqAnalysis</code> S4 class to a separate package.</li>
</ul>
</div>
<div id="deprecated-functions-and-methods" class="section level3">
<h3 class="hasAnchor">
<a href="#deprecated-functions-and-methods" class="anchor"></a>Deprecated functions and methods</h3>
<ul>
<li>Functions removed from <code>NAMESPACE</code>: magrittr pipe (<code>%&gt;%</code>), <code>DESeqAnalysis</code>, <code>DESeqResultsTables</code>, <code>alphaSummary</code>, <code>contrastName</code>, <code>export</code>, <code>markdown</code>, <code>plotDEGHeatmap</code>, <code>plotDEGPCA</code>, <code>plotMA</code>, <code>plotMeanAverage</code>, <code>plotVolcano</code>, <code>topTables</code>.</li>
<li>
<code>alphaSummary</code>: Migrated <code>DESeqDataSet</code> method to <a href="https://deseqanalysis.acidgenomcis.com/">DESeqAnalysis</a> package.</li>
</ul>
</div>
<div id="example-data" class="section level3">
<h3 class="hasAnchor">
<a href="#example-data" class="anchor"></a>Example data</h3>
<ul>
<li>Deleted <code>deseq</code> example, which has migrated to <a href="https://deseqanalysis.acidgenomcis.com/">DESeqAnalysis</a> package.</li>
</ul>
</div>
<div id="minor-changes-29" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-29" class="anchor"></a>Minor changes</h3>
<ul>
<li>Reorganized internal assert checks.</li>
<li>Miscellaneous documentation improvements.</li>
<li>Improved consistency of global formals (using <code>getOption</code>) via <code>formalsList</code> global variable.</li>
</ul>
</div>
</div>
    <div id="bcbiornaseq-0-3-3-2018-11-25" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiornaseq-0-3-3-2018-11-25" class="anchor"></a>bcbioRNASeq 0.3.3 (2018-11-25)</h2>
<div id="minor-changes-28" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-28" class="anchor"></a>Minor changes</h3>
<ul>
<li>Added additional library loads to <code>_setup.R</code>: rmarkdown, basejump, DESeq2.</li>
<li>
<code>plotGene</code>: Updated working example.</li>
<li>Improved reexport of <code>sampleData&lt;-</code> generic.</li>
<li>Updated Travis CI configuration.</li>
</ul>
</div>
</div>
    <div id="bcbiornaseq-0-3-4-2018-11-29" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiornaseq-0-3-4-2018-11-29" class="anchor"></a>bcbioRNASeq 0.3.4 (2018-11-29)</h2>
<div id="major-changes-11" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes-11" class="anchor"></a>Major changes</h3>
<ul>
<li>Improved and simplified quality control template. Removed Dropbox mode (which may be added back in a future update). Identical samples detection is now supported automatically in the plotting functions, which was previously defined in the QC template.</li>
<li>Improved subset sample handling inside extraction method. Extraction method now imports useful <code>relevelRowRanges</code> and <code>relevelColData</code> utilities from basejump.</li>
</ul>
</div>
<div id="minor-changes-27" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-27" class="anchor"></a>Minor changes</h3>
<ul>
<li>Removed <code>sampleData&lt;-</code> reexport.</li>
<li>Documentation fixes and improvements, particularly to the <code>bcbioRNASeq</code> generator function.</li>
<li>Reworked internal gffFile handling inside <code>bcbioRNASeq</code> generator function.</li>
<li>Simplified passthrough in internal <code>.new.bcbioRNASeq</code> call.</li>
<li>
<code>plot5Prime3PrimeBias</code> now uses <code>color</code> argument instead of <code>fill</code>.</li>
<li>
<code>plotDispEsts</code> now checks for duplicate samples and early returns.</li>
<li>Consistently use British spelling variants (colour instead of color) for ggplot2 functions. Our formals will stick with American English though.</li>
<li>
<code>plotTotalReads</code>: Bumped recommended default to 20 million from 10 million.</li>
</ul>
</div>
</div>
    <div id="bcbiornaseq-0-3-5-2018-12-01" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiornaseq-0-3-5-2018-12-01" class="anchor"></a>bcbioRNASeq 0.3.5 (2018-12-01)</h2>
<div id="major-changes-10" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes-10" class="anchor"></a>Major changes</h3>
<ul>
<li>Added additional improvements to quality control template. Ensuring that setup chunk never caches. bcbioRNASeq object should be called using <code>params$object_file</code>. This convention will be used across all other bcbio R packages. Reworked and simplified <code>plotCountsPerGene</code> section of QC template.</li>
</ul>
</div>
<div id="minor-changes-26" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-26" class="anchor"></a>Minor changes</h3>
<ul>
<li>Added early return method for objects without modification extract method. Might want to reconsider this approach if the user just wants to reorder samples. We’re checking for unmodified return by looking for identical raw counts matrix before and after extraction call.</li>
<li>
<code>plotCountsPerBiotype</code>: Ensure <code>normalized</code> argument uses “tpm” first.</li>
<li>
<code>plotDispEsts</code>: Switch to using <code>hasUniqueCols</code> internally instead of <code>areSamplesUnique</code>.</li>
</ul>
</div>
</div>
    <div id="bcbiornaseq-0-3-6-2018-12-12" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiornaseq-0-3-6-2018-12-12" class="anchor"></a>bcbioRNASeq 0.3.6 (2018-12-12)</h2>
<p>This is the first release that switched over to using [goalie][] for assert checks instead of [assertive][].</p>
<div id="major-changes-9" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes-9" class="anchor"></a>Major changes</h3>
<ul>
<li>
<code>bcbioRNASeq</code> generator has been overhauled to use <code>assert</code> for assert checks.</li>
</ul>
</div>
<div id="minor-changes-25" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-25" class="anchor"></a>Minor changes</h3>
<ul>
<li>Ensure title is quoted in R Markdown.</li>
<li>Switched to using <code>validate</code> in place of <code>validate_that</code> in validity checks.</li>
<li>Miscellaneous documentation improvements.</li>
<li>Now using <code>match.arg</code> internally to check validity of <code>countsFromAbundance</code> argument passthrough to <code>tximport</code>.</li>
</ul>
</div>
</div>
    <div id="bcbiornaseq-0-3-7-2019-01-13" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiornaseq-0-3-7-2019-01-13" class="anchor"></a>bcbioRNASeq 0.3.7 (2019-01-13)</h2>
<div id="major-changes-8" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes-8" class="anchor"></a>Major changes</h3>
<ul>
<li>Overhauled validity method for <code>bcbioRNASeq</code>. Improved check steps using <code>ok</code> method, similar to approach in [goalie][].</li>
<li>Added back <code>plotPCACovariates</code> method support.</li>
</ul>
</div>
<div id="minor-changes-24" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-24" class="anchor"></a>Minor changes</h3>
<ul>
<li>Updated pkgdown configuration. Removed <code>alphaSummary</code>, <code>contrastName</code>, <code>export</code>, <code>markdown</code>, <code>sampleData</code>, <code>topTables</code>
</li>
<li>[goalie][] package has been updated to support <code>nullOK</code> mode where applicable. This is particularly use for formals that use <code>NULL</code> by default. See <code>bcbioRNASeq</code> generator changes, for example.</li>
<li>
<code>prepareRNASeqTemplate</code> example is causing build checks to fail, so disable.</li>
<li>Improved CI configuration.</li>
</ul>
</div>
</div>
    <div id="bcbiornaseq-0-3-8-2019-01-17" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiornaseq-0-3-8-2019-01-17" class="anchor"></a>bcbioRNASeq 0.3.8 (2019-01-17)</h2>
<div id="minor-changes-23" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-23" class="anchor"></a>Minor changes</h3>
<ul>
<li>
<code>.tximport</code>: Use <code>makeNames</code> to check for valid names internally.</li>
<li>Removed FIXME comment and improved documentation.</li>
<li>Travis CI configuration fixes to get build checks to pass.</li>
</ul>
</div>
</div>
    <div id="bcbiornaseq-0-3-9-2019-01-23" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiornaseq-0-3-9-2019-01-23" class="anchor"></a>bcbioRNASeq 0.3.9 (2019-01-23)</h2>
<div id="minor-changes-22" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-22" class="anchor"></a>Minor changes</h3>
<ul>
<li>Bug fixes for <code>interestingGroups</code> handling in plotting functions. Needed to add back an <code>interestingGroups&lt;-</code> assignment call to slot the object.</li>
<li>Removed <code>sanitizeRowData</code> and <code>sanitizeSampleData</code> from imports.</li>
</ul>
</div>
</div>
    <div id="bcbiornaseq-0-3-10-2019-02-01" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiornaseq-0-3-10-2019-02-01" class="anchor"></a>bcbioRNASeq 0.3.10 (2019-02-01)</h2>
<div id="minor-changes-21" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-21" class="anchor"></a>Minor changes</h3>
<ul>
<li>Travis CI build configuration and documentation fixes.</li>
</ul>
</div>
</div>
    <div id="bcbiornaseq-0-3-11-2019-02-12" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiornaseq-0-3-11-2019-02-12" class="anchor"></a>bcbioRNASeq 0.3.11 (2019-02-12)</h2>
<div id="minor-changes-20" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-20" class="anchor"></a>Minor changes</h3>
<ul>
<li>Renamed <code>plotGene</code> to <code>plotCounts</code>, to match consistency in <a href="https://basejump.acidgenomics.com/">basejump</a> and <a href="https://deseqanalysis.acidgenomcis.com/">DESeqAnalysis</a> package updates.</li>
<li>Updated dependencies to provide improved backwards compatibility support for R 3.4 / Bioconductor 3.6.</li>
</ul>
</div>
</div>
    <div id="bcbiornaseq-0-3-12-2019-03-18" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiornaseq-0-3-12-2019-03-18" class="anchor"></a>bcbioRNASeq 0.3.12 (2019-03-18)</h2>
<div id="major-changes-7" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes-7" class="anchor"></a>Major changes</h3>
<ul>
<li>
<code>sampleData</code>: Moved method support to <a href="https://basejump.acidgenomics.com/">basejump</a> package.</li>
</ul>
</div>
<div id="minor-changes-19" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-19" class="anchor"></a>Minor changes</h3>
<ul>
<li>Resaved example data.</li>
<li>Sample metadata chunk in quality control template now returns <code>DataFrame</code>.</li>
<li>Improved some deprecation messages.</li>
<li>CI configuration improvements.</li>
</ul>
</div>
</div>
    <div id="bcbiornaseq-0-3-13-2019-03-29" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiornaseq-0-3-13-2019-03-29" class="anchor"></a>bcbioRNASeq 0.3.13 (2019-03-29)</h2>
<div id="minor-changes-18" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-18" class="anchor"></a>Minor changes</h3>
<ul>
<li>Added additional checks to <code>bcbioRNASeq</code> validity method.</li>
<li>Draft update to quality control template, to support parameterized <code>bcbioRNASeq</code> object input. This is likely not necessary and will be removed in a future update.</li>
<li>Added additional checks for <code>bcbioRNASeq</code> to <code>DESeqDataSet</code> coercion.</li>
<li>Improved internal <code>perMillion</code> argument code handling inside plotting functions: <code>plotMappedReads</code>, <code>plotTotalReads</code>.</li>
<li>Removed <code>reexports.R</code>. No longer reexporting <code>Gene2Symbol</code> or magrittr pipe.</li>
</ul>
</div>
</div>
    <div id="bcbiornaseq-0-3-14-2019-04-01" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiornaseq-0-3-14-2019-04-01" class="anchor"></a>bcbioRNASeq 0.3.14 (2019-04-01)</h2>
<p>Moved forked development repository to [Acid Genomics][].</p>
<div id="major-changes-6" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes-6" class="anchor"></a>Major changes</h3>
<ul>
<li>Switched unit testing data URL from seq.cloud to acidgenomics.com.</li>
</ul>
</div>
<div id="minor-changes-17" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-17" class="anchor"></a>Minor changes</h3>
<ul>
<li>Reworked internal reexports for assignment generics. See <code>counts</code> for example.</li>
<li>Miscellaneous documentation improvements.</li>
<li>Switched from <code>basejump</code> prefix to <code>acid</code>, where applicable.</li>
<li>Reexported essential functions from basejump required for QC template to render without basejump being attached.</li>
</ul>
</div>
</div>
    <div id="bcbiornaseq-0-3-15-2019-04-11" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiornaseq-0-3-15-2019-04-11" class="anchor"></a>bcbioRNASeq 0.3.15 (2019-04-11)</h2>
<div id="minor-changes-16" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-16" class="anchor"></a>Minor changes</h3>
<ul>
<li>Ensuring that all <a href="https://basejump.acidgenomics.com/">basejump</a> functions necessary for the quality control template get reexported.</li>
<li>Renamed R Markdown template files and <code>travis-render.sh</code> script into kebab case, rather than snake case.</li>
<li>Documentation improvements (using <code>inheritParams</code>).</li>
</ul>
</div>
</div>
    <div id="bcbiornaseq-0-3-16-2019-04-17" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiornaseq-0-3-16-2019-04-17" class="anchor"></a>bcbioRNASeq 0.3.16 (2019-04-17)</h2>
<div id="minor-changes-15" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-15" class="anchor"></a>Minor changes</h3>
<ul>
<li>Tweaks to quality control template.</li>
<li>Switched Travis CI configuration to use <code>rnaseq</code> Docker image.</li>
</ul>
</div>
</div>
    <div id="bcbiornaseq-0-3-17-2019-04-22" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiornaseq-0-3-17-2019-04-22" class="anchor"></a>bcbioRNASeq 0.3.17 (2019-04-22)</h2>
<div id="minor-changes-14" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-14" class="anchor"></a>Minor changes</h3>
<ul>
<li>Bar plots now show the first sample at the top of the Y axis when <code>flip = TRUE</code>. This is more human readable.</li>
<li>Quality control template now uses <code>bcbiornaseq_file</code> as main input param.</li>
<li>Bug fix: added back internal <code>.normalizedTrans</code> function required for some quality control plots.</li>
</ul>
</div>
</div>
    <div id="bcbiornaseq-0-3-18-2019-04-25" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiornaseq-0-3-18-2019-04-25" class="anchor"></a>bcbioRNASeq 0.3.18 (2019-04-25)</h2>
<div id="minor-changes-13" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-13" class="anchor"></a>Minor changes</h3>
<ul>
<li>S4 generic reexport documentation fixes.</li>
</ul>
</div>
</div>
    <div id="bcbiornaseq-0-3-19-2019-05-05" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiornaseq-0-3-19-2019-05-05" class="anchor"></a>bcbioRNASeq 0.3.19 (2019-05-05)</h2>
<div id="major-changes-5" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes-5" class="anchor"></a>Major changes</h3>
<ul>
<li>Now pinned to R &gt;= 3.5.</li>
</ul>
</div>
<div id="minor-changes-12" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-12" class="anchor"></a>Minor changes</h3>
<ul>
<li>Improved Travis CI and AppVeyor CI configuration.</li>
</ul>
</div>
</div>
    <div id="bcbiornaseq-0-3-20-2019-05-29" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiornaseq-0-3-20-2019-05-29" class="anchor"></a>bcbioRNASeq 0.3.20 (2019-05-29)</h2>
<div id="minor-changes-11" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-11" class="anchor"></a>Minor changes</h3>
<ul>
<li>Added support for size factor adjusted counts as “sf” in <code>counts</code> call.</li>
<li>Improved internal trans axis labeling handling for DESeqTransform counts.</li>
<li>
<code>plotMeanSD</code>: Improved default colors. Improved internal non-zero count handling. Reworked plot headers.</li>
</ul>
</div>
</div>
    <div id="bcbiornaseq-0-3-21-2019-07-17" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiornaseq-0-3-21-2019-07-17" class="anchor"></a>bcbioRNASeq 0.3.21 (2019-07-17)</h2>
<div id="major-changes-4" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes-4" class="anchor"></a>Major changes</h3>
<ul>
<li>Initial support for <code>bcbioRNASeq</code> to <code>DGEList</code> (edgeR) object coercion.</li>
<li>
<code>relativeLogExpression</code>: Improved internal code, using edgeR <code>calcNormFactors</code> approach.</li>
</ul>
</div>
<div id="minor-changes-10" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-10" class="anchor"></a>Minor changes</h3>
<ul>
<li>Updated basejump dependency to v0.10.11.</li>
<li>Improved Travis CI docker config to check coverage and lints.</li>
<li>Improved Bioconductor installation instructions.</li>
</ul>
</div>
</div>
    <div id="bcbiornaseq-0-3-22-2019-07-23" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiornaseq-0-3-22-2019-07-23" class="anchor"></a>bcbioRNASeq 0.3.22 (2019-07-23)</h2>
<div id="major-changes-3" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes-3" class="anchor"></a>Major changes</h3>
<ul>
<li>
<code>bcbioRNASeq</code>: Now supporting <code>fast</code> argument, which enables the user to skip any internal DESeq2 calculations. This will skip generation of <code>normalized</code>, <code>vst</code>, and <code>fpkm</code> matrices. Note that automatic <code>rlog</code> matrix generation is no longer supported, since it’s often too slow, especially for large datasets. If you want to generate rlog counts, simply coerce the bcbioRNASeq object to a DESeqDataSet, then run the <code>rlog</code> function.</li>
<li>Aligned counts are now saved inside the object when a pseudoaligner (e.g. salmon, kallisto) are used to define the primary <code>counts</code> assay. These values are saved in an <code>aligned</code> matrix.</li>
</ul>
</div>
<div id="new-functions" class="section level3">
<h3 class="hasAnchor">
<a href="#new-functions" class="anchor"></a>New functions</h3>
<ul>
<li>
<code>pseudoVsAligned</code>: Visually inspect the correlation of pseudoaligned counts to the aligned counts. Note that default bcbio genomes use legacy UCSC builds, where as the pseudoaligners generate values from newer Ensembl transcript annotations.</li>
</ul>
</div>
<div id="minor-changes-9" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-9" class="anchor"></a>Minor changes</h3>
<ul>
<li>Updated package dependency versions.</li>
</ul>
</div>
</div>
    <div id="bcbiornaseq-0-3-23-2019-07-29" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiornaseq-0-3-23-2019-07-29" class="anchor"></a>bcbioRNASeq 0.3.23 (2019-07-29)</h2>
<div id="major-changes-2" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes-2" class="anchor"></a>Major changes</h3>
<ul>
<li>
<code>updateObject</code>: Worked internal approach to be compatible with upcoming Bioconductor 3.10 release. Has to rethink the strategy to handling assays internally, which can no longer be easily coerced from assays slot in the <code>bcbioRNASeq</code> object, due to changes in <code>ShallowSimpleListAssays</code> handling. This should return an object in exactly the same manner as previous releases.</li>
</ul>
</div>
</div>
    <div id="bcbiornaseq-0-3-24-2019-08-12" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiornaseq-0-3-24-2019-08-12" class="anchor"></a>bcbioRNASeq 0.3.24 (2019-08-12)</h2>
<div id="minor-changes-8" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-8" class="anchor"></a>Minor changes</h3>
<ul>
<li>Simplified internal object creation in <code>bcbioRNASeq</code>, <code><a href="https://rdrr.io/r/base/Extract.html">[</a></code>, and <code>updateObject</code>.</li>
<li>Improved message consistency.</li>
</ul>
</div>
</div>
    <div id="bcbiornaseq-0-3-25-2019-08-20" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiornaseq-0-3-25-2019-08-20" class="anchor"></a>bcbioRNASeq 0.3.25 (2019-08-20)</h2>
<div id="major-changes-1" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes-1" class="anchor"></a>Major changes</h3>
<ul>
<li>Made subsetting factor for extract (<code><a href="https://rdrr.io/r/base/Extract.html">[</a></code>) operations.</li>
<li>
<code>updateObject</code>: Improved <code>rowRanges</code> handling.</li>
</ul>
</div>
<div id="minor-changes-7" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-7" class="anchor"></a>Minor changes</h3>
<ul>
<li>
<code>bcbioRNASeq</code> generator now captures call using <code>standardizeCall</code> instead of <code>match.call</code>, and automatically expands all default arguments.</li>
<li>Improved NAMESPACE and basejump dependencies.</li>
<li>Made vsn and DEGreport packages optional, to reduce the number of dependencies required for installation.</li>
<li>
<code>plotPCACovariates</code>: Reworked internal code to not depend on dplyr.</li>
<li>Removed internal code dependencies on magrittr pipe.</li>
</ul>
</div>
</div>
    <div id="bcbiornaseq-0-3-26-2019-08-27" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiornaseq-0-3-26-2019-08-27" class="anchor"></a>bcbioRNASeq 0.3.26 (2019-08-27)</h2>
<div id="minor-changes-6" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-6" class="anchor"></a>Minor changes</h3>
<ul>
<li>Now requiring R 3.6, as we near a candidate Bioconductor release.</li>
<li>Updated basejump dependencies.</li>
</ul>
</div>
</div>
    <div id="bcbiornaseq-0-3-27-2019-09-09" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiornaseq-0-3-27-2019-09-09" class="anchor"></a>bcbioRNASeq 0.3.27 (2019-09-09)</h2>
<div id="minor-changes-5" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-5" class="anchor"></a>Minor changes</h3>
<ul>
<li>
<code>bcbioRNASeq</code>: Internal generator now calls <code>importSampleData</code> using <code>pipeline = "bcbio"</code> argument, so we don’t run into breaking changes when handling user metadata in a future basejump update.</li>
<li>Updated basejump dependency versions.</li>
</ul>
</div>
</div>
    <div id="bcbiornaseq-0-3-28-2019-09-17" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiornaseq-0-3-28-2019-09-17" class="anchor"></a>bcbioRNASeq 0.3.28 (2019-09-17)</h2>
<div id="minor-changes-4" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-4" class="anchor"></a>Minor changes</h3>
<ul>
<li>Reworked quality control plots to use “labels” instead of “title”.</li>
<li>Updated R Markdown templates and added render code coverage in testthat-extra.</li>
<li>Updated vignette to use shared bibliography.bib file from basejump.</li>
</ul>
</div>
</div>
    <div id="bcbiornaseq-0-3-29-2019-10-30" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiornaseq-0-3-29-2019-10-30" class="anchor"></a>bcbioRNASeq 0.3.29 (2019-10-30)</h2>
<div id="minor-changes-3" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-3" class="anchor"></a>Minor changes</h3>
<ul>
<li>Added support for Bioconductor 3.10.</li>
<li>Updated internal <code>DataFrame</code> validity checks to also check for <code>DFrame</code> class, which has changed in latest S4Vectors 0.23 release.</li>
<li>Updated basejump dependency versions.</li>
</ul>
</div>
</div>
    <div id="bcbiornaseq-0-3-30-2020-01-20" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiornaseq-0-3-30-2020-01-20" class="anchor"></a>bcbioRNASeq 0.3.30 (2020-01-20)</h2>
<div id="major-changes" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes" class="anchor"></a>Major changes</h3>
<ul>
<li>Improved internal handling of <code>bcbioRNASeq</code> to <code>DGEList</code> coercion, for handoff to edgeR and limma-voom. Now the code handles <code>countsFromAbundance</code> settings more intelligently and informs the user whether downstream edgeR and/or limma-voom handoff is appropriate.</li>
<li>Now exporting <code>as.DESeqDataSet</code>, <code>as.DESeqTransform</code>, and <code>as.DGEList</code> S4 generic methods, which support arguments. Note that quick coercion using only <code>as</code> (see <code>coerce</code> documentation from Bioconductor) does not support arguments.</li>
</ul>
</div>
<div id="minor-changes-2" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-2" class="anchor"></a>Minor changes</h3>
<ul>
<li>Fixed support for bcbio fastrnaseq pipeline. Thanks to <a href='https://github.com/kokyriakidis'>@kokyriakidis</a> for pointing this out. Added code coverage support for an example run produced via this pipeline.</li>
<li>Updated basejump dependencies, namely renaming bioverbs to acidgenerics.</li>
<li>Now using cli package for improved messages.</li>
</ul>
</div>
</div>
    <div id="bcbiornaseq-0-3-31-2020-02-19" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiornaseq-0-3-31-2020-02-19" class="anchor"></a>bcbioRNASeq 0.3.31 (2020-02-19)</h2>
<div id="minor-changes-1" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-1" class="anchor"></a>Minor changes</h3>
<ul>
<li>Relaxed validity checks against <code>dataVersions</code> and <code>programsVersions</code> in <code>metadata</code> slot, for compatibility with <code>updateObject</code> upgrade path from v0.2.9 objects, which are currently generated by bcbio-nextgen. Thanks <a href='https://github.com/sjmonkley'>@sjmonkley</a> for catching this. See issue <a href='https://github.com/hbc/bcbioRNASeq/issues/146'>#146</a>.</li>
<li>Changed license from MIT to GPL-3.</li>
<li>Updated documentation.</li>
</ul>
</div>
</div>
    <div id="bcbiornaseq-0-3-32-2020-02-24" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiornaseq-0-3-32-2020-02-24" class="anchor"></a>bcbioRNASeq 0.3.32 (2020-02-24)</h2>
<div id="minor-changes" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes" class="anchor"></a>Minor changes</h3>
<ul>
<li>
<code>bcbioRNASeq</code>: Removed now defunct <code>spikeNames</code> argument. Refer to <code>makeSummarizedExperiment</code> documentation and release notes in basejump package for details.</li>
</ul>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <div id="tocnav">
      <h2>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
        <li><a href="#bcbiornaseq-0-3-0-2018-11-06">0.3.0</a></li>
        <li><a href="#bcbiornaseq-0-3-1-2018-11-14">0.3.1</a></li>
        <li><a href="#bcbiornaseq-0-3-2-2018-11-19">0.3.2</a></li>
        <li><a href="#bcbiornaseq-0-3-3-2018-11-25">0.3.3</a></li>
        <li><a href="#bcbiornaseq-0-3-4-2018-11-29">0.3.4</a></li>
        <li><a href="#bcbiornaseq-0-3-5-2018-12-01">0.3.5</a></li>
        <li><a href="#bcbiornaseq-0-3-6-2018-12-12">0.3.6</a></li>
        <li><a href="#bcbiornaseq-0-3-7-2019-01-13">0.3.7</a></li>
        <li><a href="#bcbiornaseq-0-3-8-2019-01-17">0.3.8</a></li>
        <li><a href="#bcbiornaseq-0-3-9-2019-01-23">0.3.9</a></li>
        <li><a href="#bcbiornaseq-0-3-10-2019-02-01">0.3.10</a></li>
        <li><a href="#bcbiornaseq-0-3-11-2019-02-12">0.3.11</a></li>
        <li><a href="#bcbiornaseq-0-3-12-2019-03-18">0.3.12</a></li>
        <li><a href="#bcbiornaseq-0-3-13-2019-03-29">0.3.13</a></li>
        <li><a href="#bcbiornaseq-0-3-14-2019-04-01">0.3.14</a></li>
        <li><a href="#bcbiornaseq-0-3-15-2019-04-11">0.3.15</a></li>
        <li><a href="#bcbiornaseq-0-3-16-2019-04-17">0.3.16</a></li>
        <li><a href="#bcbiornaseq-0-3-17-2019-04-22">0.3.17</a></li>
        <li><a href="#bcbiornaseq-0-3-18-2019-04-25">0.3.18</a></li>
        <li><a href="#bcbiornaseq-0-3-19-2019-05-05">0.3.19</a></li>
        <li><a href="#bcbiornaseq-0-3-20-2019-05-29">0.3.20</a></li>
        <li><a href="#bcbiornaseq-0-3-21-2019-07-17">0.3.21</a></li>
        <li><a href="#bcbiornaseq-0-3-22-2019-07-23">0.3.22</a></li>
        <li><a href="#bcbiornaseq-0-3-23-2019-07-29">0.3.23</a></li>
        <li><a href="#bcbiornaseq-0-3-24-2019-08-12">0.3.24</a></li>
        <li><a href="#bcbiornaseq-0-3-25-2019-08-20">0.3.25</a></li>
        <li><a href="#bcbiornaseq-0-3-26-2019-08-27">0.3.26</a></li>
        <li><a href="#bcbiornaseq-0-3-27-2019-09-09">0.3.27</a></li>
        <li><a href="#bcbiornaseq-0-3-28-2019-09-17">0.3.28</a></li>
        <li><a href="#bcbiornaseq-0-3-29-2019-10-30">0.3.29</a></li>
        <li><a href="#bcbiornaseq-0-3-30-2020-01-20">0.3.30</a></li>
        <li><a href="#bcbiornaseq-0-3-31-2020-02-19">0.3.31</a></li>
        <li><a href="#bcbiornaseq-0-3-32-2020-02-24">0.3.32</a></li>
      </ul>
    </div>
  </div>

</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://steinbaugh.com/'>Michael Steinbaugh</a>, <a href='https://lpantano.github.io/'>Lorena Pantano</a>, <a href='https://github.com/roryk/'>Rory Kirchner</a>, <a href='https://github.com/vbarrera/'>Victor Barrera</a>, <a href='https://github.com/chapmanb/'>Brad Chapman</a>, <a href='https://github.com/marypiper/'>Mary Piper</a>, <a href='https://github.com/mistrm82/'>Meeta Mistry</a>, <a href='https://github.com/rkhetani/'>Radhika Khetani</a>, <a href='https://github.com/KayleighRutherford/'>Kayleigh Rutherford</a>, <a href='https://github.com/ohofmann/'>Oliver Hofmann</a>, <a href='https://github.com/jnhutchinson/'>John Hutchinson</a>, <a href='https://github.com/sjhosui/'>Shannan Ho Sui</a>, <a href='https://bioinformatics.sph.harvard.edu/'>Harvard Chan Bioinformatics Core</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


