<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Extract or replace parts of an object — extract • bcbioRNASeq</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  

<meta property="og:title" content="Extract or replace parts of an object — extract" />
<meta property="og:description" content="Extract genes by row and samples by column." />
<meta name="twitter:card" content="summary" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">bcbioRNASeq</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.31</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/bcbioRNASeq.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/hbc/bcbioRNASeq">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Extract or replace parts of an object</h1>
    <small class="dont-index">Source: <a href='https://github.com/hbc/bcbioRNASeq/blob/master/R/extract-methods.R'><code>R/extract-methods.R</code></a></small>
    <div class="hidden name"><code>extract.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Extract genes by row and samples by column.</p>
    </div>

    <pre class="usage"># S4 method for bcbioRNASeq,ANY,ANY,ANY
[(x, i, j, drop = FALSE, recalculate = TRUE)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>x</th>
      <td><p>object from which to extract element(s) or in which to replace element(s).</p></td>
    </tr>
    <tr>
      <th>i</th>
      <td><p>indices specifying elements to extract or replace.  Indices are
    <code>numeric</code> or <code>character</code> vectors or empty (missing) or
    <code>NULL</code>.  Numeric values are coerced to integer as by
    <code><a href='https://rdrr.io/r/base/integer.html'>as.integer</a></code> (and hence truncated towards zero).
    Character vectors will be matched to the <code><a href='https://rdrr.io/r/base/names.html'>names</a></code> of the
    object (or for matrices/arrays, the <code><a href='https://rdrr.io/r/base/dimnames.html'>dimnames</a></code>):
    see &#8216;Character indices&#8217; below for further details.</p>
<p>For <code>[</code>-indexing only: <code>i</code>, <code>j</code>, <code>...</code> can be
    logical vectors, indicating elements/slices to select.  Such vectors
    are recycled if necessary to match the corresponding extent.
    <code>i</code>, <code>j</code>, <code>...</code> can also be negative integers,
    indicating elements/slices to leave out of the selection.</p>
<p>When indexing arrays by <code>[</code> a single argument <code>i</code> can be a
    matrix with as many columns as there are dimensions of <code>x</code>; the
    result is then a vector with elements corresponding to the sets of
    indices in each row of <code>i</code>.</p>
<p>An index value of <code>NULL</code> is treated as if it were <code><a href='https://rdrr.io/r/base/integer.html'>integer(0)</a></code>.</p></td>
    </tr>
    <tr>
      <th>j</th>
      <td><p>indices specifying elements to extract or replace.  Indices are
    <code>numeric</code> or <code>character</code> vectors or empty (missing) or
    <code>NULL</code>.  Numeric values are coerced to integer as by
    <code><a href='https://rdrr.io/r/base/integer.html'>as.integer</a></code> (and hence truncated towards zero).
    Character vectors will be matched to the <code><a href='https://rdrr.io/r/base/names.html'>names</a></code> of the
    object (or for matrices/arrays, the <code><a href='https://rdrr.io/r/base/dimnames.html'>dimnames</a></code>):
    see &#8216;Character indices&#8217; below for further details.</p>
<p>For <code>[</code>-indexing only: <code>i</code>, <code>j</code>, <code>...</code> can be
    logical vectors, indicating elements/slices to select.  Such vectors
    are recycled if necessary to match the corresponding extent.
    <code>i</code>, <code>j</code>, <code>...</code> can also be negative integers,
    indicating elements/slices to leave out of the selection.</p>
<p>When indexing arrays by <code>[</code> a single argument <code>i</code> can be a
    matrix with as many columns as there are dimensions of <code>x</code>; the
    result is then a vector with elements corresponding to the sets of
    indices in each row of <code>i</code>.</p>
<p>An index value of <code>NULL</code> is treated as if it were <code><a href='https://rdrr.io/r/base/integer.html'>integer(0)</a></code>.</p></td>
    </tr>
    <tr>
      <th>drop</th>
      <td><p>For matrices and arrays.  If <code>TRUE</code> the result is
    coerced to the lowest possible dimension (see the examples).  This
    only works for extracting elements, not for the replacement.  See
    <code><a href='https://rdrr.io/r/base/drop.html'>drop</a></code> for further details.</p></td>
    </tr>
    <tr>
      <th>recalculate</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code>.
Recalculate DESeq2 normalized counts and variance-stabilizing
transformations defined in <code>assays</code>. Recommended by default, but can take a
long time for large datasets. If <code>FALSE</code>, these assays will be removed
automatically: <code>normalized</code>, <code>rlog</code>, <code>vst</code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p><code>bcbioRNASeq</code>.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Internal count transformations are rescaled automatically, if defined. DESeq2
transformations will only be updated when <code>recalculate = TRUE</code>.</p>
    <h2 class="hasAnchor" id="note"><a class="anchor" href="#note"></a>Note</h2>

    <p>Updated 2020-01-20.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Becker, R. A., Chambers, J. M. and Wilks, A. R. (1988)
  <em>The New S Language</em>.
  Wadsworth &amp; Brooks/Cole.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='no'>bcb</span>)
<span class='no'>object</span> <span class='kw'>&lt;-</span> <span class='no'>bcb</span>

<span class='co'>## Minimum of 100 genes, 2 samples.</span>
<span class='no'>genes</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span>(<span class='no'>object</span>), <span class='fl'>100L</span>)
<span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span>(<span class='no'>genes</span>)</div><div class='output co'>#&gt; [1] "ENSMUSG00000000001" "ENSMUSG00000000003" "ENSMUSG00000000028"
#&gt; [4] "ENSMUSG00000000049" "ENSMUSG00000000058" "ENSMUSG00000000078"</div><div class='input'><span class='no'>samples</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span>(<span class='no'>object</span>), <span class='fl'>2L</span>)
<span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span>(<span class='no'>samples</span>)</div><div class='output co'>#&gt; [1] "control_rep1" "control_rep2"</div><div class='input'>
<span class='co'>## Extract by sample name.</span>
<span class='no'>object</span>[, <span class='no'>samples</span>]</div><div class='output co'>#&gt; <span class='message'>→ Recalculating <span style='color: #0000BB;'>DESeq2</span><span> normalizations.</span></div><div class='output co'>#&gt; <span class='message'>Generating </span><span style='color: #000000;background-color: #BBBBBB;'>`DESeqDataSet`</span><span> with </span><span style='color: #0000BB;'>DESeq2</span><span> 1.26.0.</span></div><div class='output co'>#&gt; <span class='message'>estimating size factors</span></div><div class='output co'>#&gt; <span class='message'>estimating dispersions</span></div><div class='output co'>#&gt; <span class='message'>gene-wise dispersion estimates</span></div><div class='output co'>#&gt; <span class='message'>mean-dispersion relationship</span></div><div class='output co'>#&gt; <span class='message'>final dispersion estimates</span></div><div class='output co'>#&gt; <span class='message'>fitting model and testing</span></div><div class='output co'>#&gt; <span class='message'>→ </span><span style='color: #000000;background-color: #BBBBBB;'>`varianceStabilizingTransformation()`</span><span></span></div><div class='output co'>#&gt; bcbioRNASeq 0.3.27
#&gt; uploadDir: /data00/draco/acidbase/packages/bcbioRNASeq/inst/extdata/bcbio
#&gt; dates(2): [bcbio] 2018-03-18; [R] 2019-09-16
#&gt; level: genes
#&gt; caller: salmon
#&gt; organism: Mus musculus
#&gt; interestingGroups(2): treatment day
#&gt; class: RangedSummarizedExperiment 
#&gt; dim: 100 2 
#&gt; metadata(28): allSamples bcbioCommandsLog ... yaml subset
#&gt; assays(7): counts aligned ... vst fpkm
#&gt; rownames(100): ENSMUSG00000000001 ENSMUSG00000000003 ...
#&gt;   ENSMUSG00000062661 ENSMUSG00000074340
#&gt; rowData names(8): broadClass description ... geneName seqCoordSystem
#&gt; colnames(2): control_rep1 control_rep2
#&gt; colData names(26): averageInsertSize averageReadLength ... treatment
#&gt;   x5x3Bias</div><div class='input'>
<span class='co'>## Extract by gene list.</span>
<span class='no'>object</span>[<span class='no'>genes</span>, ]</div><div class='output co'>#&gt; bcbioRNASeq 0.3.27
#&gt; uploadDir: /data00/draco/acidbase/packages/bcbioRNASeq/inst/extdata/bcbio
#&gt; dates(2): [bcbio] 2018-03-18; [R] 2019-09-16
#&gt; level: genes
#&gt; caller: salmon
#&gt; organism: Mus musculus
#&gt; interestingGroups(2): treatment day
#&gt; class: RangedSummarizedExperiment 
#&gt; dim: 100 6 
#&gt; metadata(27): allSamples bcbioCommandsLog ... wd yaml
#&gt; assays(7): counts aligned ... vst fpkm
#&gt; rownames(100): ENSMUSG00000000001 ENSMUSG00000000003 ...
#&gt;   ENSMUSG00000062661 ENSMUSG00000074340
#&gt; rowData names(8): broadClass description ... geneName seqCoordSystem
#&gt; colnames(6): control_rep1 control_rep2 ... fa_day7_rep2 fa_day7_rep3
#&gt; colData names(26): averageInsertSize averageReadLength ... treatment
#&gt;   x5x3Bias</div><div class='input'>
<span class='co'>## Extract by both genes and samples.</span>
<span class='no'>x</span> <span class='kw'>&lt;-</span> <span class='no'>object</span>[<span class='no'>genes</span>, <span class='no'>samples</span>]</div><div class='output co'>#&gt; <span class='message'>→ Recalculating </span><span style='color: #0000BB;'>DESeq2</span><span> normalizations.</span></div><div class='output co'>#&gt; <span class='message'>Generating </span><span style='color: #000000;background-color: #BBBBBB;'>`DESeqDataSet`</span><span> with </span><span style='color: #0000BB;'>DESeq2</span><span> 1.26.0.</span></div><div class='output co'>#&gt; <span class='message'>estimating size factors</span></div><div class='output co'>#&gt; <span class='message'>estimating dispersions</span></div><div class='output co'>#&gt; <span class='message'>gene-wise dispersion estimates</span></div><div class='output co'>#&gt; <span class='message'>mean-dispersion relationship</span></div><div class='output co'>#&gt; <span class='message'>final dispersion estimates</span></div><div class='output co'>#&gt; <span class='message'>fitting model and testing</span></div><div class='output co'>#&gt; <span class='message'>→ </span><span style='color: #000000;background-color: #BBBBBB;'>`varianceStabilizingTransformation()`</span><span></span></div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/print.html'>print</a></span>(<span class='no'>x</span>)</div><div class='output co'>#&gt; bcbioRNASeq 0.3.27
#&gt; uploadDir: /data00/draco/acidbase/packages/bcbioRNASeq/inst/extdata/bcbio
#&gt; dates(2): [bcbio] 2018-03-18; [R] 2019-09-16
#&gt; level: genes
#&gt; caller: salmon
#&gt; organism: Mus musculus
#&gt; interestingGroups(2): treatment day
#&gt; class: RangedSummarizedExperiment 
#&gt; dim: 100 2 
#&gt; metadata(28): allSamples bcbioCommandsLog ... yaml subset
#&gt; assays(7): counts aligned ... vst fpkm
#&gt; rownames(100): ENSMUSG00000000001 ENSMUSG00000000003 ...
#&gt;   ENSMUSG00000062661 ENSMUSG00000074340
#&gt; rowData names(8): broadClass description ... geneName seqCoordSystem
#&gt; colnames(2): control_rep1 control_rep2
#&gt; colData names(26): averageInsertSize averageReadLength ... treatment
#&gt;   x5x3Bias</div><div class='input'><span class='fu'>assayNames</span>(<span class='no'>x</span>)</div><div class='output co'>#&gt; [1] "counts"      "aligned"     "avgTxLength" "tpm"         "normalized" 
#&gt; [6] "vst"         "fpkm"       </div><div class='input'>
<span class='co'>## Fast subsetting, by skipping DESeq2 recalculations.</span>
<span class='co'>## Note that `normalized`, `rlog`, and `vst` assays will be removed.</span>
<span class='no'>x</span> <span class='kw'>&lt;-</span> <span class='no'>object</span>[, <span class='no'>samples</span>, <span class='kw'>recalculate</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>]</div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BBBB00;'>!</span><span> Skipping </span><span style='color: #000000;background-color: #BBBBBB;'>`DESeq2()`</span><span> normalizations.</span></div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/print.html'>print</a></span>(<span class='no'>x</span>)</div><div class='output co'>#&gt; bcbioRNASeq 0.3.27
#&gt; uploadDir: /data00/draco/acidbase/packages/bcbioRNASeq/inst/extdata/bcbio
#&gt; dates(2): [bcbio] 2018-03-18; [R] 2019-09-16
#&gt; level: genes
#&gt; caller: salmon
#&gt; organism: Mus musculus
#&gt; interestingGroups(2): treatment day
#&gt; class: RangedSummarizedExperiment 
#&gt; dim: 100 2 
#&gt; metadata(28): allSamples bcbioCommandsLog ... yaml subset
#&gt; assays(4): counts aligned avgTxLength tpm
#&gt; rownames(100): ENSMUSG00000000001 ENSMUSG00000000003 ...
#&gt;   ENSMUSG00000062661 ENSMUSG00000074340
#&gt; rowData names(8): broadClass description ... geneName seqCoordSystem
#&gt; colnames(2): control_rep1 control_rep2
#&gt; colData names(26): averageInsertSize averageReadLength ... treatment
#&gt;   x5x3Bias</div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span>(<span class='fu'>assays</span>(<span class='no'>x</span>))</div><div class='output co'>#&gt; [1] "counts"      "aligned"     "avgTxLength" "tpm"        </div></span></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      <li><a href="#value">Value</a></li>
      <li><a href="#details">Details</a></li>
      <li><a href="#note">Note</a></li>
      <li><a href="#references">References</a></li>
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    <p>Michael Steinbaugh, Lorena Pantano</p>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://steinbaugh.com/'>Michael Steinbaugh</a>, <a href='https://lpantano.github.io/'>Lorena Pantano</a>, <a href='https://github.com/roryk/'>Rory Kirchner</a>, <a href='https://github.com/vbarrera/'>Victor Barrera</a>, <a href='https://github.com/chapmanb/'>Brad Chapman</a>, <a href='https://github.com/marypiper/'>Mary Piper</a>, <a href='https://github.com/mistrm82/'>Meeta Mistry</a>, <a href='https://github.com/rkhetani/'>Radhika Khetani</a>, <a href='https://github.com/KayleighRutherford/'>Kayleigh Rutherford</a>, <a href='https://github.com/ohofmann/'>Oliver Hofmann</a>, <a href='https://github.com/jnhutchinson/'>John Hutchinson</a>, <a href='https://github.com/sjhosui/'>Shannan Ho Sui</a>, <a href='https://bioinformatics.sph.harvard.edu/'>Harvard Chan Bioinformatics Core</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


